test/geocoders/banfrance.py

```diff
@@ -0,0 +1,53 @@
+from abc import ABCMeta, abstractmethod
+
+from six import with_metaclass
+
+from geopy.geocoders import banfrance
+from test.geocoders.util import GeocoderTestBase
+
+
+class BaseBANFranceTestCase(with_metaclass(ABCMeta, object)):
+
+    delta = 0.04
+
+    @classmethod
+    @abstractmethod
+    def make_geocoder(cls, **kwargs):
+        pass
+
+    @classmethod
+    def setUpClass(cls):
+        cls.geocoder = cls.make_geocoder()
+
+    def test_geocode_with_address(self):
+        """
+        banfrance.geocode Address
+        """
+        self.geocode_run(
+            {"query": "Camp des Landes, 41200 VILLEFRANCHE-SUR-CHER"},
+            {"latitude": 47.293048,
+                "longitude": 1.718985,
+                "address": "Le Camp des Landes 41200 Villefranche-sur-Cher"},
+        )
+
+    def test_reverse(self):
+        """
+        banfrance.reverse simple
+        """
+        res = self.reverse_run(
+            {"query": '48.154587,3.221237',
+                "exactly_one": True},
+            {},
+        )
+        self.assertEqual(
+            res.address,
+            '15 Rue des Fontaines 89100 Collemiers'
+        )
+
+
+class BANFranceTestCase(BaseBANFranceTestCase, GeocoderTestBase):
```

Comment:
There's no need in an abstract class for this geocoder, as it is one of its kind (unlike Nominatim). Could you please use a single class, like, for example, [test/geocoders/yandex.py](https://github.com/geopy/geopy/blob/3b19c8e4f6fdf13dddfbe7a3703ba031ec95194d/test/geocoders/yandex.py)?